src/Domain/CoordinateMaps/TimeDependentHelpers.hpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace domain {
+/// Check if the call to the coordinate map is time-dependent
+template <typename T>
+using is_map_time_dependent_t = tt::is_callable_t<
+    T, std::array<std::decay_t<T>, std::decay_t<T>::dim>, double,
+    std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>>;
+
+/// Check if the call to the coordinate map is time-dependent
+template <typename T>
+constexpr bool is_map_time_dependent_v = is_map_time_dependent_t<T>::value;
+
+namespace detail {
+CREATE_IS_CALLABLE(jacobian)
+}  // namespace detail
+
+/// Check if the call to the Jacobian of the coordinate map is time-dependent
```

Comment:
I'm a bit confused by your question. This check should work for all coordinate maps that we will ever write, no need for more type traits (maybe Hessian since we will probably need those). Does that clarify things?

Reply:
Let me see if I understand: This metafunction is keyed to examining the `jacobian` member function to determine if it is time dependent. Do I understand correctly that we'll only want to check the `jacobian` and not the `inverse` or `inv_jacobian`, but maybe a Hessian which could be addressed in later additions?

Reply:
@moxcodes if either the map or Jacobian are time-dependent, the corresponding inverses must also be so we need only one type trait for each...  and similarly for the Hessians when they are added

Reply:
might be worth adding the above to the doxygen for the type traits